### PR TITLE
Ruby: improve diagnostic messages

### PR DESCRIPTION
**Pull request by aibaars**

### CODE REVIEW

File: ruby/extractor/src/diagnostics.rs [hunk 1]
```diff
@@ -199,9 +199,25 @@ impl DiagnosticLoggers {
     }
 }
 
+fn longest_backtick_sequence_length(text: &str) -> usize {
```
Comment (tausbn) [inline ruby/extractor/src/diagnostics.rs:202]: I can't help but feel that what this function is called is not in line with what it actually calculates. Perhaps `backtick_count` is a better name for what it does?

Comment (aibaars) [reply ruby/extractor/src/diagnostics.rs:202]: Yes, good  point. The function is not right.

File: ruby/extractor/src/extractor.rs [hunk 2]
```diff
@@ -383,15 +401,13 @@ impl<'a> Visitor<'a> {
...
                 self.record_parse_error(
                     loc,
                     self.diagnostics_writer
-                        .message("parse-error", "Parse error")
+                        .new_entry("parse-error", "Parse error")
                         .severity(diagnostics::Severity::Error)
                         .location(self.path, start_line, start_column, end_line, end_column)
-                        .text(&error_message)
-                        .status_page(),
+                        .message("unknown table type: {}", &[node.kind()]),
```
Comment (calumgrant) [inline ruby/extractor/src/extractor.rs:410]: I notice an inconsistency in the capitalization of the message text - is there a convention or style guide to refer to about this?

**Review by subatoi — DISMISSED** [review]

**Review by erik-krogh — APPROVED** [review]
I'm not that confortable in Rust, and I had to read up on what e.g. `with_capacity` does.  

But everything I understand, I approve of :shipit:  

You are doing more kinds of diagnostics messages than I had expected.

**Pull request merged**